matrix.md

```diff
@@ -0,0 +1,136 @@
+| **Developer Experience** | | |
+| Debugging support | Standard tools / Custom tools / Limited | |
+| Failure modes visibility | How failures manifest (crashes / logs / error codes / silent failures) | |
+| **Availability** | | |
+| Usage model / licensing | Open source / Commercial / Research / Restricted / Other: ___ | |
+| Ease of access | Easy to obtain / Requires approval / Difficult / Other: ___ | |
+
+---
+
+## Composability
+
+| **Dimension** | **Metric** | **Value / Notes** |
+|----------------|------------|-------------------|
+| **Cross-Framework Composability** | | |
+| Works with VMs/containers | ✅ / ❌ / Partial | |
+| Works with other compartmentalization systems | ✅ / ❌ (if no, specify which conflict: ___) | |
+| Can be layered/stacked | ✅ / ❌ | |
+| **Inter-Compartment Interactions** | | |
+| How compartments invoke each other | Function calls / Message passing / RPC / Syscalls / Other: ___ | |
+| How compartments share data | Shared memory / Message copying / Serialization / Other: ___ | |
+| Interaction semantics | Synchronous / Asynchronous / Both | |
+| Interaction security/validation | [Describe: e.g., "Capability-based interface", "Type-checked", "Manual validation", "None"] | |
+| **Decomposition Model** | | |
+| Typical compartment boundary | Library / Service / Thread / Process / VM / Other: ___ | |
+| Who defines boundaries | Programmer / Compiler / Runtime / Kernel / Hardware | |
+| Boundaries flexible at runtime | ✅ / ❌ | |
+| **System Integration** | | |
+| Threading model | Works with standard threads / Requires custom threading / Other: ___ | |
+| Process model | Works with fork/exec / Custom / N/A | |
+| POSIX compatibility | Full / Partial / Limited | |
+| **Composition Limitations** | | |
+| Known limitations when composed | [Describe: e.g., "Performance degrades 2x when inside VMs", "Cannot nest with other MPK users"] | |
+| Security caveats when layered | [Describe: e.g., "Outer layer can bypass inner protections"] | |
+
+---
+
+## System Design & Operations
+
+| **Dimension** | **Metric** | **Value / Notes** |
+|----------------|------------|-------------------|
+| **Target Environment** | | |
+| Primary deployment target | Embedded / IoT / Cloud / Server / Desktop / Other: ___ | |
+| Deployment scale | Single device / Cluster / Large scale | |
+| **Integration** | | |
+| Monitoring/orchestration support | Good / Limited / None | |
+| CI/CD compatibility | Good / Limited / Custom | |
+| **Operations** | | |
+| Configuration complexity | Low / Medium / High | |
+| Maintenance burden | Low / Medium / High | |
+
+---
+
+## Cost & Resources
+
+| **Dimension** | **Metric** | **Value / Notes** |
+|----------------|------------|-------------------|
+| CPU/memory overhead | Low / Medium / High | |
+| Requires specialized hardware | ✅ / ❌ (if yes: ___) | |
+| License/operational costs | Free / Moderate / High / Enterprise | |
```

Comment:
This also feels redundant with usage model and ease of access in Usability / Adoptability above.